tests/ws_integration.rs

```diff
@@ -235,6 +236,155 @@ mod ws_integration {
 		Ok(())
 	}
 
+	#[test(tokio::test)]
+	async fn variable_auth_live_query() -> Result<(), Box<dyn std::error::Error>> {
+		let (addr, _server) = common::start_server_with_defaults().await.unwrap();
+		let socket = &mut common::connect_ws(&addr).await?;
+
+		//
+		// Prepare the connection
+		//
+		let res = common::ws_signin(socket, USER, PASS, None, None, None).await;
+		assert!(res.is_ok(), "result: {:?}", res);
+		let res = common::ws_use(socket, Some("N"), Some("D")).await;
+		assert!(res.is_ok(), "result: {:?}", res);
+
+		// Setup scope
+		let res = common::ws_query(socket, r#"
+        DEFINE SCOPE scope SESSION 2s
+            SIGNUP ( CREATE user SET email = $email, pass = crypto::argon2::generate($pass) )
+            SIGNIN ( SELECT * FROM user WHERE email = $email AND crypto::argon2::compare(pass, $pass) )
+        ;"#).await;
+		assert!(res.is_ok(), "result: {:?}", res);
+
+		// Signup
+		let res = common::ws_send_msg_and_wait_response(
+			socket,
+			serde_json::to_string(&json!({
+				"id": "1",
+				"method": "signup",
+				"params": [{
+					"ns": "N",
+					"db": "D",
+					"sc": "scope",
+					"email": "email@email.com",
+					"pass": "pass",
+				}],
+			}))
+			.unwrap(),
+		)
+		.await;
+		assert!(res.is_ok(), "result: {:?}", res);
+
+		// Sign in
+		let res = common::ws_send_msg_and_wait_response(
```

Comment:
you can use `ws_signin` here, see line 78 for an example

Reply:
if you use the `ws_signin` method, you can get rid of all the verifications below, because the method is already tested by other tests